src/v/archival/archival_metadata_stm.h

```diff
@@ -133,7 +135,8 @@ class archival_metadata_stm final : public raft::persisted_stm<> {
       raft::consensus*,
       cloud_storage::remote& remote,
       features::feature_table&,
-      ss::logger& logger);
+      ss::logger& logger,
+      std::optional<cloud_storage::remote_label>);
```

Comment:
Default to `std::nullopt`? Or would you prefer constructors be explicit about the `remote_label`?

Reply:
Yea, i'd prefer constructors here be explicit, at least for now. It helps prove that we're not missing some plumbing